test/index.spec.ts

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { ApolloServer } from 'apollo-server';
+import { AppDataSource } from '../src/data-source';
+import { expect } from 'chai';
+import { resolvers } from '../src/graphql/resolvers';
+import { typeDefs } from '../src/graphql/schema';
+
+describe('query hello', function () {
+  before('Starting server', function (done) {
+    AppDataSource.initialize()
+      .then(() => {
+        const server = new ApolloServer({ typeDefs, resolvers });
+        server.listen().then(({ url }) => {
+          console.log(`Server listening at ${url}`);
+        });
+      })
+      .catch((error) => console.log(error))
+      .then(done);
+  });
+  it('Returning hello', async function () {
```

Comment:
Outros pontos para reflexão, principalmente pensando no futuro:

1. Esse before está muito semelhante ao setup que vc faz no código principal do servidor, acho que valeria a pena reaproveitar, né?

2. O seu script de teste prevê a execução de vários arquivos que terminem em `.spec.ts`. No entanto seria interessante que antes da execução dos testes houvesse o setup do servidor, correto? No caso de mais arquivos de teste, não faria muito sentido esse setup ser feito em vários befores de forma repetida, dado que ele poderia ser feito uma vez só antes de todos os testes. Depois dá uma olhada no vídeo de setup dos testes, lá tem uma sugestão simples pra isso.

Reply:
Fiz essas alterações no commit "Fix: Start the server once for testing"